nasa-gibs/worldview
# WV-2680: Overhauling Dataset Collections Process for Multi-Day Products

## Description

Now that GITC flipped to the cloud there are many multi-day products that include layer-identifier-actual headers to display the dataset versions (NRT vs STD). Our current setup for updating the dataset collections did not account for multi-day products. We are pulling the headers from the tileLoadFunction in the layerBuilder process which is only queried once per layer period. This means, for example, that the collection datasets were not be updated for each day for a 4-day product.

This PR updates the dataset collections process so that the querying is now done in its own React component triggered on date change and other layer events. This allows us to query the collection headers on appropriate date changes (daily/subdaily). This also allows us to mitigate making requests for headers that have already been queried. 

To view the current behavior in production, you can use this [link](https://worldview.earthdata.nasa.gov/?v=-351.93426107210996,-179.5383270711262,274.6238556348794,121.66643424166088&l=Coastlines_15m,MODIS_Combined_L4_FPAR_4Day,MODIS_Combined_L4_LAI_4Day,MODIS_Terra_L3_NDVI_Monthly,MODIS_Terra_L3_Land_Surface_Temp_Monthly_Day,VIIRS_SNPP_CorrectedReflectance_TrueColor&lg=true&t=2023-10-01-T16%3A04%3A05Z). Roll the date back by one day at a time and observe how the dataset collection badges will appear and disappear.

## How To Test

1. `git pull wv-2680`
2. `npm ci`
3. `npm run watch`
4. [Open WV]([http://localhost:3000/?v=-351.93426107210996,-179.5383270711262,274.6238556348794,121.66643424166088&l=Coastlines_15m,MODIS_Combined_L4_FPAR_4Day,MODIS_Combined_L4_LAI_4Day,MODIS_Terra_L3_NDVI_Monthly,MODIS_Terra_L3_Land_Surface_Temp_Monthly_Day,VIIRS_SNPP_CorrectedReflectance_TrueColor&lg=true&t=2023-10-01-T16%3A04%3A05Z](https://worldview.earthdata.nasa.gov/?v=-351.93426107210996,-179.5383270711262,274.6238556348794,121.66643424166088&l=Coastlines_15m,MODIS_Combined_L4_FPAR_4Day,MODIS_Combined_L4_LAI_4Day,MODIS_Terra_L3_NDVI_Monthly,MODIS_Terra_L3_Land_Surface_Temp_Monthly_Day,VIIRS_SNPP_CorrectedReflectance_TrueColor&lg=true&t=2023-10-01-T16%3A04%3A05Z))
5. Move the date back incrementally by day and observe that the dataset collections are updated on each date

@nasa-gibs/worldview
